src/RequirementChecker/IO.php

```diff
@@ -0,0 +1,281 @@
+<?php
+
+/*
+ * This file is part of the box project.
+ *
+ * (c) Kevin Herrera <kevin@herrera.io>
+ *     Théo Fidry <theo.fidry@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace KevinGH\Box\RequirementChecker;
+
+/**
+ * The code in this file must be PHP 5.3+ compatible as is used to know if the application can be run.
+ *
+ * @private
+ */
+final class IO
+{
+    const VERBOSITY_QUIET = 16;
+    const VERBOSITY_NORMAL = 32;
+    const VERBOSITY_VERBOSE = 64;
+    const VERBOSITY_VERY_VERBOSE = 128;
+    const VERBOSITY_DEBUG = 256;
+    
+    private $tokens;
+    private $parsed;
+    private $interactive;
+    private $verbosity = self::VERBOSITY_NORMAL;
+    private $colorSupport;
+
+    public function __construct()
+    {
+        $this->parseInput();
+
+        $shellVerbosity = $this->configureVerbosity();
+        $this->interactive = $this->checkInteractivity($shellVerbosity);
+        $this->colorSupport = $this->checkColorSupport();
+    }
+
+    /**
+     * @return bool
+     */
+    public function isInteractive()
+    {
+        return $this->interactive;
+    }
+
+    /**
+     * @return int
+     */
+    public function getVerbosity()
+    {
+        return $this->verbosity;
+    }
+
+    /**
+     * @param int $verbosity
+     */
+    public function setVerbosity($verbosity)
+    {
+        $this->verbosity = $verbosity;
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasColorSupport()
+    {
+        return $this->colorSupport;
+    }
+
+    /**
+     * @return array
+     */
+    private function parseInput()
+    {
+        $this->tokens = $_SERVER['argv'];
```

Comment:
I agree the arguments management is not as complex as a regular input and is bound to be simpler because it can't know about the input definitions. So I simplified the implementation to reflect on that.

I admit I'm not familiar enough with the `getopt()` function. I added the tests for the IO class so it should be easier to change whilst ensuring the same feature set.

Would it be possible for you to take a look at it?

Reply:
After a quick test `getopt` have some issue for this case:

 - It can not be _UnitTested_ (cf. [this message](https://stackoverflow.com/a/23913181))
 - It can not handle option `-vv` and `-vvv`

Reply:
Too bad, well I hope the current implementation appears simpler than previously then